sdk/txbuild/formation.go

```diff
@@ -0,0 +1,42 @@
+package txbuild
+
+import (
+	"github.com/stellar/go/keypair"
+	"github.com/stellar/go/txnbuild"
+)
+
+func Formation(initiator *keypair.FromAddress, responder *keypair.FromAddress, initiatorEscrow *keypair.FromAddress, responderEscrow *keypair.FromAddress, startSequence int64) (*txnbuild.Transaction, error) {
+	tp := txnbuild.TransactionParams{
+		SourceAccount: &txnbuild.SimpleAccount{
+			AccountID: initiatorEscrow.Address(),
+			Sequence:  startSequence,
+		},
+		BaseFee:    txnbuild.MinBaseFee,
+		Timebounds: txnbuild.NewTimeout(300),
+	}
+	tp.Operations = append(tp.Operations, &txnbuild.BeginSponsoringFutureReserves{SourceAccount: initiator.Address(), SponsoredID: initiatorEscrow.Address()})
+	tp.Operations = append(tp.Operations, &txnbuild.SetOptions{
+		SourceAccount:   initiatorEscrow.Address(),
+		MasterWeight:    txnbuild.NewThreshold(0),
+		LowThreshold:    txnbuild.NewThreshold(2),
+		MediumThreshold: txnbuild.NewThreshold(2),
+		HighThreshold:   txnbuild.NewThreshold(2),
+		Signer:          &txnbuild.Signer{Address: responder.Address(), Weight: 1},
```

Comment:
should the formation transaction also require the initiator to be added as a signer to EI (and responder to ER) with weight of 1? Eg. if before `F` is submitted `I` changes their signing weight to 0 on `EI`, then `F` is submitted. That would lock `EI` right? Or `I` changes their signing weight to 2

another option could be part of validating F for responder is checking EI has initiator with weight 1.

Reply:
That's a good point. This assumes I and R haven't messed things up by removing themselves as signers. Which is not so bad if the initial contributions are zero, but if they are zero, it is bad. You could trap someone that way. I'll change it here and in the SEP too.